sdk/aqueduct/integrations/dynamic_k8s_integration.py

```diff
@@ -1,10 +1,38 @@
+from typing import Dict, Union
+
+from aqueduct.backend.response_models import DynamicEngineStatusResponse
 from aqueduct.constants.enums import K8sClusterActionType, K8sClusterStatusType
-from aqueduct.error import InvalidIntegrationException
+from aqueduct.error import InvalidIntegrationException, InvalidUserArgumentException
+from aqueduct.integrations.connect_config import DynamicK8sConfig
 from aqueduct.models.integration import Integration, IntegrationInfo
 
 from aqueduct import globals
 
 
+def parse_dynamic_k8s_config(
+    config_delta: Union[Dict[str, Union[int, str]], DynamicK8sConfig]
+) -> DynamicK8sConfig:
+    if not isinstance(config_delta, dict) and not isinstance(config_delta, DynamicK8sConfig):
+        raise InvalidUserArgumentException(
+            "`config_delta` argument must be either a dict or DynamicK8sConfig."
+        )
```

Comment:
Would it be possible to typecheck the possible keys here? The pydantic error messaging is really bad so it might be nice for us to give them some help here.

Reply:
I'm trying to just see if `config_delta = DynamicK8sConfig(**config_delta, extra="ignore")` works. If so, even if they provide extra keys, we should be fine.

Reply:
Actually, even without the `extra="ignore"`, I think the error message looks fine: if you have extra field it'll just say extra field not permitted. Also if the type of each field is wrong, the pydantic serialization will succeed but the Golang side will catch the deserialization error, which is also fine, so I'll keep the current implementation now.